tools/clang/lib/SPIRV/AstTypeProbe.cpp

```diff
@@ -624,6 +624,22 @@ bool isSameScalarOrVecType(QualType type1, QualType type2) {
   return false;
 }
 
+bool isSameScalarOrVecMatType(QualType type1, QualType type2) {
```

Comment:
I'm not sure if this function is really needed. It repeats the logic in `isSameType`. We could just use `isSameType`?

Reply:
removed.